File: python/leetcode/206-reverse-linkedlist.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        current = head
        previous, after = None, None

        if current is None or current.next is None:
            return head

        while current.next is not None:
            temp = current.next
            current.next = previous
            previous = current
            current = temp

        # you'll end up at last node
        head = current
        head.next = previous

        return head

File: c#/leetcode/206-reverse-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        ListNode current = head;
        ListNode prev = null;
        
        
        while (current != null) {
            ListNode next = current.next;
            if (next == null) {
                head = current;
            }
            current.next = prev;
            prev = current;
            current = next;
        }
        
        return head;
    }
}